### DISCUSSION TITLE
Admin Page is ok but nothing else

### DISCUSSION THREAD

**Discussion by DERIKKU-GER** [dani-garcia/vaultwarden#3610] [General]
![image](https://github.com/dani-garcia/vaultwarden/assets/82025456/72e176a0-e70f-4b0c-8311-687066bd8d34)

nothing is going. Only admin pages are ok

this is in the log:
[2023-06-23 10:04:27.316][start][INFO] Rocket has launched from http://0.0.0.0:80

[2023-06-23 10:04:39.503][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:39.504][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:40.938][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:40.939][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:40.939][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:40.945][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:41.931][rocket::server::_][WARN] Remote left: channel closed.

[2023-06-23 10:04:41.934][rocket::server::_][WARN] Remote left: channel closed.

### Your environment (Generated via diagnostics page)
* Vaultwarden version: v1.28.1
* Web-vault version: v2023.3.0b
* OS/Arch: linux/x86_64
* Running within Docker: true (Base: Debian)
* Environment settings overridden: true
* Uses a reverse proxy: true
* IP Header check: true (X-Real-IP)
* Internet access: true
* Internet access via a proxy: false
* DNS Check: true
* Browser/Server Time Check: true
* Server/NTP Time Check: true
* Domain Configuration Check: true
* HTTPS Check: true
* Database type: SQLite
* Database version: 3.39.2
* Clients used: 
* Reverse proxy and version: 
* Other relevant information: 

### Config (Generated via diagnostics page)
<details><summary>Show Running Config</summary>

**Environment settings which are overridden:** ADMIN_TOKEN

```json
{
  "_duo_akey": null,
  "_enable_duo": true,
  "_enable_email_2fa": false,
  "_enable_smtp": true,
  "_enable_yubico": true,
  "_icon_service_csp": "",
  "_icon_service_url": "",
  "_ip_header_enabled": true,
  "_smtp_img_src": "cid:",
  "admin_ratelimit_max_burst": 3,
  "admin_ratelimit_seconds": 300,
  "admin_session_lifetime": 20,
  "admin_token": "***",
  "allowed_iframe_ancestors": "",
  "attachments_folder": "data/attachments",
  "authenticator_disable_time_drift": false,
  "data_folder": "data",
  "database_conn_init": "",
  "database_max_conns": 10,
  "database_timeout": 30,
  "database_url": "***************",
  "db_connection_retries": 15,
  "disable_2fa_remember": false,
  "disable_admin_token": false,
  "disable_icon_download": false,
  "domain": "*****://*****************************",
  "domain_origin": "*****://*****************************",
  "domain_path": "",
  "domain_set": true,
  "duo_host": null,
  "duo_ikey": null,
  "duo_skey": null,
  "email_attempts_limit": 3,
  "email_expiration_time": 600,
  "email_token_size": 6,
  "emergency_access_allowed": true,
  "emergency_notification_reminder_schedule": "0 3 * * * *",
  "emergency_request_timeout_schedule": "0 7 * * * *",
  "enable_db_wal": true,
  "event_cleanup_schedule": "0 10 0 * * *",
  "events_days_retain": null,
  "extended_logging": true,
  "helo_name": null,
  "hibp_api_key": null,
  "icon_blacklist_non_global_ips": true,
  "icon_blacklist_regex": null,
  "icon_cache_folder": "data/icon_cache",
  "icon_cache_negttl": 259200,
  "icon_cache_ttl": 2592000,
  "icon_download_timeout": 10,
  "icon_redirect_code": 302,
  "icon_service": "internal",
  "incomplete_2fa_schedule": "30 * * * * *",
  "incomplete_2fa_time_limit": 3,
  "invitation_expiration_hours": 120,
  "invitation_org_name": "Vaultwarden",
  "invitations_allowed": true,
  "ip_header": "X-Real-IP",
  "job_poll_interval_ms": 30000,
  "log_file": null,
  "log_level": "Info",
  "log_timestamp_format": "%Y-%m-%d %H:%M:%S.%3f",
  "login_ratelimit_max_burst": 10,
  "login_ratelimit_seconds": 60,
  "org_attachment_limit": null,
  "org_creation_users": "",
  "org_events_enabled": false,
  "org_groups_enabled": false,
  "password_hints_allowed": true,
  "password_iterations": 600000,
  "reload_templates": false,
  "require_device_email": false,
  "rsa_key_filename": "data/rsa_key",
  "send_purge_schedule": "0 5 * * * *",
  "sendmail_command": null,
  "sends_allowed": true,
  "sends_folder": "data/sends",
  "show_password_hint": false,
  "signups_allowed": true,
  "signups_domains_whitelist": "",
  "signups_verify": false,
  "signups_verify_resend_limit": 6,
  "signups_verify_resend_time": 3600,
  "smtp_accept_invalid_certs": false,
  "smtp_accept_invalid_hostnames": false,
  "smtp_auth_mechanism": null,
  "smtp_debug": false,
  "smtp_embed_images": true,
  "smtp_explicit_tls": null,
  "smtp_from": "",
  "smtp_from_name": "Vaultwarden",
  "smtp_host": null,
  "smtp_password": null,
  "smtp_port": 587,
  "smtp_security": "starttls",
  "smtp_ssl": null,
  "smtp_timeout": 15,
  "smtp_username": null,
  "templates_folder": "data/templates",
  "tmp_folder": "data/tmp",
  "trash_auto_delete_days": 30,
  "trash_purge_schedule": "0 5 0 * * *",
  "use_sendmail": false,
  "use_syslog": false,
  "user_attachment_limit": null,
  "web_vault_enabled": true,
  "web_vault_folder": "web-vault/",
  "websocket_address": "0.0.0.0",
  "websocket_enabled": false,
  "websocket_port": 3012,
  "yubico_client_id": null,
  "yubico_secret_key": null,
  "yubico_server": null
}
```
</details>

**Comment by stefan0xC** [comment]
Can you check if you have disk space left?

**Reply by DERIKKU-GER** [reply]
thanks, but thats no  problem. the data dir is on my home dir with more than 2tb free

**Comment by BlackDex** [comment]
It could be a browser extension which is blocking some javascript loading.

What do you see in the browser developer console (F12)?

**Reply by DERIKKU-GER** [reply]
Failed to load resource: net::ERR_HTTP2_PROTOCOL_ERROR
main.3a27378a743dd4ad9f70.js:1 
        
        
        Failed to load resource: net::ERR_HTTP2_PROTOCOL_ERROR
scriptlets.js:308  Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'wasm-unsafe-eval'". Either the 'unsafe-inline' keyword, a hash ('sha256-IwIK2LOKYOzjtZsuCJZoxuuQa0qjgu5Vt7tf1ddGePU='), or a nonce ('nonce-...') is required to enable inline execution.

injectScriptlet @ scriptlets.js:308
scriptlets.js:308  Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'wasm-unsafe-eval'". Either the 'unsafe-inline' keyword, a hash ('sha256-NPd/muSZkiykJmPa9AWMv9BCXlvJqIXjpRFKND1jfdQ='), or a nonce ('nonce-...') is required to enable inline execution.

injectScriptlet @ scriptlets.js:308
scriptlets.js:308  Refused to execute inline script because it violates the following Content Security Policy directive: "script-src 'self' 'wasm-unsafe-eval'". Either the 'unsafe-inline' keyword, a hash ('sha256-6slwZsNfzkpqOq0NnEvsvfZkccp/EHcnG5RmbfLiPqo='), or a nonce ('nonce-...') is required to enable inline execution.

injectScriptlet @ scriptlets.js:308
(Index):1957  crbug/1173575, non-JS module files deprecated.

**Reply by BlackDex** [reply]
You probably override the security headers, or some other addon/extension of the reverse proxy you use is doing that.
Vaultwarden needs specific security headers, and if they do not match, it will fail.
It is best to let Vaultwarden handle these security headers and not change them to ensure all clients and all features work as intended.

**Reply by DERIKKU-GER** [reply]
>>It is best to let Vaultwarden handle these security headers and not change them to ensure all clients and all features work as intended

well but how i can du this. i have nothing change an my swag proxy or some on my server

**Reply by BlackDex** [reply]
I have no clue. it's your setup.

Vaultwarden it self sends all the right headers, so if the reverse proxy just passes those on, it should not be an issue.
So, verify your reverse proxy setup, and see if there is anything there which could override these headers.
ModSecurity, WAF, or anything else in regards to security.

**Comment by xBaaDf00d** [comment]
I also have this problem, Vault Warden is in a Vault Warden Docker Container on a linux host, running on a different port mapped to 80 in the container.
I have No Extensions installed, And I get the Spinning Wheel.
If Vault warden is pulling something from an external Source My router could be blocking Ad Services, that is it.
I can not see what Is wrong with my Docker -- vaultwarden/server:latest -- Image being used.

Oh and its on an 8TB disk, with about 6tb free So its not a diskspace issue. 

'''
Unhandled error in angular Error: Could not instantiate WebCryptoFunctionService. Could not locate Subtle crypto.
    useClass web-crypto-function.service.ts:17
    i r3_injector.ts:620
    hydrate r3_injector.ts:492
    get r3_injector.ts:341
    Xe injector_compatibility.ts:79
    Je injector_compatibility.ts:112
    nt injector_compatibility.ts:327
    i r3_injector.ts:620
    hydrate r3_injector.ts:492
    get r3_injector.ts:341
    Xe injector_compatibility.ts:79
    Je injector_compatibility.ts:112
    nt injector_compatibility.ts:327
    i r3_injector.ts:620
    hydrate r3_injector.ts:492
'''

**Reply by BlackDex** [reply]
You need https!
https://github.com/dani-garcia/vaultwarden/wiki/Enabling-HTTPS

**Reply by xBaaDf00d** [reply]
Thanks for the info, as it turns out I run my Own CA for other apps, I'll apply this to VW, and get it sorted thanks.

Just logically though, why is the default docker container running on port 80? and not 443? with an error in the log like need cert?  If it requires a certificate to run. shouldn't port 80 be disabled by default?

Thanks again for the information I'll be able to get it working from that link.

**Reply by BlackDex** [reply]
Most people run it via a reverse proxy and might not want or need to encrypt the traffic also between the reverse proxy and the endpoint application.

Also, Vaultwarden runs just fine on http on localhost/127.0.0.1.

**Comment by sehranrasooljan** [comment]
I was facing the same issue. To fix it, I set up Cloudflare as a proxy and added my domain. This allowed me to access Vaultwarden over port 443 (HTTPS) instead of port 80, and the issue was resolved.
